### ISSUE TITLE
Create an alfresco-jar-archetype

### ISSUE THREAD

**Issue by mindthegab** [Alfresco/alfresco-sdk#173]
_From [maurizio...@alfresco.com](https://code.google.com/u/112978353022216972888/) on December 12, 2013 12:52:15_

Based on the alfresco-amp-archetype-1.1.1, I performed the following experiment; I haven't tested it intensively, but surefire tests and tomcat run work.

If there are no objections (warnings or unsupported use cases), I'd like to proceed with the development of an alfresco-jar-archetype.
1. Changed packaging from amp to jar
2. Moved src/main/amp/config into src/main/resources
3. Added the following XML snippet to pom.xml
   
   <build>
     <plugins>
       <plugin>
         <groupId>org.alfresco.maven.plugin</groupId>
         <artifactId>alfresco-maven-plugin</artifactId>
         <version>${maven.alfresco.version}</version>
         <extensions>true</extensions>
         <configuration>
           <skipAmpInstallation>true</skipAmpInstallation>
         </configuration>
         <executions>
           <execution>
             <id>resolve-snapshot-version</id>
             <phase>initialize</phase>
             <goals>
               <goal>set-version</goal>
             </goals>
           </execution>
         </executions>
       </plugin>
     </plugins>
   </build>

_Original issue: http://code.google.com/p/maven-alfresco-archetypes/issues/detail?id=173_

**Comment by mindthegab** [comment]
We are picking back this effort, looking to certify JAR modules / extensions in the next version. Sample project (mutation of the AMP archetype) here:

https://github.com/Alfresco/alfresco-sdk-samples/tree/alfresco-simple-module/samples/alfresco-simple-module/

NOTE: This is currently an investigation / testing phase, so please use at your own risk and for development purposes only (not in production).

**Comment by mindthegab** [comment]
JAR support for repo and Share is planned for the next version, therefore we need to provide archetypes and remove lots of pointless logic in the process.

Also JAR modules will be loaded from an external folder, so we need a separate issue for that.

**Comment by binduwavell** [comment]
In Zia-Alfresco-QuickStart we create a jar module and an amp module for each repo and share extension group. Until Alfresco supports deploying jars with immutable wars, I suspect this would make sense for the SDK. AMP becomes an almost completely empty shell for injecting the awesome jars into the war. Only really need one AMP for repo and one for share, in theory these can deploy all har dependencies in one go. Once the new loader stuff is done, the amp modules can be deleted. 

I feel that a single module that produces a jar and then wraps it with an amp wrapper an abuse of maven. While, I concede that is does keep the project structure more simple...

**Comment by ohej** [comment]
SDK3 only provides JAR archetypes out of the box. AMPs are still available as an optional assembly. Closing
